### ISSUE TITLE
Cannot run wpscan after updating Kali

### ISSUE THREAD

**Issue by J4C0814N** [wpscanteam/wpscan#1243]
### Subject of the issue
I cannot run wpscan after updating Kali last night. I get a gem dependency error with the activesupport gem.

### Your environment
* Version of WPScan: Cannot determine, wpscan wont run
* Version of Ruby: Ruby 2.5.1p57 (2018-03-29 revision 63029) [i386-linux-gnu]
* Operating System (OS): Kali 4.18.0-kali2-686 Debian 4.18.10-2kali1

### Steps to reproduce
`sudo apt-get update && sudo apt-get upgrade && sudo apt-get dist-upgrade`
try to run `wpscan`

### Expected behavior
wpscan runs.

### Actual behavior
Any attempts to run wpscan threw the following error:
```
Traceback (most recent call last):
	16: from /usr/local/bin/wpscan:23:in '<main>'
	15: from /usr/lib/ruby/2.5.0/rubygems.rb:309:in 'activate_bin_path'
	14: from /usr/lib/ruby/2.5.0/rubygems.rb:309:in 'synchronize'
	13: from /usr/lib/ruby/2.5.0/rubygems.rb:310:in 'block in activate_bin_path'
	12: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in 'activate'
	11: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in 'activate_dependencies'
	10: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in 'each'
	 9: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1472:in 'block in activate_dependencies'
	 8: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in 'activate'
	 7: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in 'activate_dependencies'
	 6: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in 'each'
	 5: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1472:in 'block in activate_dependencies'
	 4: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in 'activate'
	 3: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in 'activate_dependencies'
	 2: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in 'each'
	 1: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1469:in 'block in activate_dependencies'
/usr/lib/ruby/2.5.0/rubygems/dependency.rb:312:in 'to_specs': Could not find 'activesupport' (~> 4.2.1) - did find: [activesupport-5.2.1] (Gem::MissingSpecVersionError)
Checked in 'GEM_PATH=/root/.gem/ruby/2.5.0:/var/lib/gems/2.5.0:/usr/lib/i386-linux-gnu/rubygems-integration/2.5.0:/usr/share/rubygems-integration/2.5.0:/usr/share/rubygems-integration/all', execute 'gem env' for more information
```

So I installed activesupport 4.2.1 with `gem install activesupport -v 4.2.1`
and now when I try to run wpscan I get the following error:
```
Traceback (most recent call last):
	11: from /usr/local/bin/wpscan:23:in '<main>'
	10: from /usr/lib/ruby/2.5.0/rubygems.rb:309:in 'activate_bin_path'
	 9: from /usr/lib/ruby/2.5.0/rubygems.rb:309:in 'synchronize'
	 8: from /usr/lib/ruby/2.5.0/rubygems.rb:310:in 'block in activate_bin_path'
	 7: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in 'activate'
	 6: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in 'activate_dependencies'
	 5: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in 'each'
	 4: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1472:in 'block in activate_dependencies'
	 3: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in 'activate'
	 2: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in 'activate_dependencies'
	 1: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in 'each'
/usr/lib/ruby/2.5.0/rubygems/specification.rb:1466:in 'block in activate_dependencies': can't satisfy 'activesupport (~> 5.2)', already activated 'activesupport-4.2.1' (Gem::LoadError)
```

### What have you already tried
Tried updating Ruby through apt-get
`gem update`
`gem install wpscan`
downloading the git repo and running `bundle install && rake install`
`gem install activesupport -v 4.2.1`

Things you have tried (where relevant):

* Update WPScan to the latest version [X ]
* Update Ruby to the latest version [X ]
* Ensure you can reach the target site using cURL [X ]
* Proxied WPScan through a HTTP proxy to view the raw traffic [ ]
* Ensure you are using a supported Operating System (Linux and macOS) [X]

**Comment by ethicalhack3r** [comment]
Assuming you're running wpscan with just `wpscan` in the terminal?

**Comment by erwanlr** [comment]
@FireFart I don't like such specific things, we did it in the v2 but people still tried to install whatever rather than open a bug issue on the Kali tracker.

Here I don't get the error, it says require activesupport 4.2.1 even though the ~> 5.2 is in the gemspec, and then when the 4.2.1 is installed, it says that the 4.2.1 is already loaded and can't load the 5.2.x. Something went clearly wrong after updating Kali it seems

**Comment by FireFart** [comment]
@erwanlr jeah but people tend to make it worse by installing the gems manually :/

**Comment by J4C0814N** [comment]
@ethicalhack3r Yes this happens when trying to run `wpscan` directly from the terminal, with any or no flags.

@FireFart and @erwanlr Should I uninstall and re-install Ruby, then re-isntall wpscan?
I am not a Ruby developer, but I would not have thought installing the version of a gem it was complaining about would break it. (not at wpscan, I mean Ruby). I install, remove and update python modules all the time and it does not affect the python environment this heavily. 

This could be a well known quirk of Ruby, but if you didn't know better, the messaging , to me at least, indicated a simple fix of "Just install version 4.2.1 of activesupport". That is why I performed those steps.

**Comment by FireFart** [comment]
normally this messes things up in kali if you try to install gems as everything is bundled in kali together. So you can try reinstalling wpscan via apt again and see if this fixes the error

**Comment by Excoroxus** [comment]
can anyone solve the problem ?

**Comment by parteeksingh005** [comment]
Subject of the issue:
I cannot run wpscan after updating Kali last night. I get a gem dependency error with the activesupport gem.

My Kali environment:
    Version of WPScan: Cannot determine, wpscan wont run
    Version of Ruby: ruby 2.5.1p57 (2018-03-29 revision 63029) [x86_64-linux-gnu]
    Operating System (OS): Linux kali 4.18.0-kali2-amd64 #1 SMP Debian 4.18.10-2kali1 (2018-10-09) x86_64 GNU/Linux

Expected behavior:
wpscan runs.

Actual behavior:
Any attempts to run wpscan threw the following error:
Traceback (most recent call last):
	11: from /usr/local/bin/wpscan:23:in `<main>'
	10: from /usr/lib/ruby/2.5.0/rubygems.rb:309:in `activate_bin_path'
	 9: from /usr/lib/ruby/2.5.0/rubygems.rb:309:in `synchronize'
	 8: from /usr/lib/ruby/2.5.0/rubygems.rb:310:in `block in activate_bin_path'
	 7: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in `activate'
	 6: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `activate_dependencies'
	 5: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `each'
	 4: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1472:in `block in activate_dependencies'
	 3: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in `activate'
	 2: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `activate_dependencies'
	 1: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `each'
/usr/lib/ruby/2.5.0/rubygems/specification.rb:1466:in `block in activate_dependencies': can't satisfy 'activesupport (~> 5.2)', already activated 'activesupport-4.2.10' (Gem::LoadError)
anyone can help me out with this issue?

**Comment by WouterVanVegchel** [comment]
I have the same issue

**Comment by ethicalhack3r** [comment]
Can someone who has the issue open an issue on the Kali bug tracker (https://bugs.kali.org/view_all_bug_page.php) or forum (https://forums.kali.org/forum.php) and link to it from here, please?

**Comment by sumgro** [comment]
Following... 

Getting the same error on Kali, when triggering the command wpscan.

**Comment by FireFart** [comment]
@sumgro as @ethicalhack3r said it's an kali issue, not a wpscan issue. Can you please open a bug on the kali bug tracker and link it here? https://bugs.kali.org/main_page.php

**Comment by sumgro** [comment]
@FireFart created a ticket on kali bug tracker - https://bugs.kali.org/view.php?id=5120

**Comment by sumgro** [comment]
this seems to have been caused by clash of 2 processes of installing via gem and through kali repositories...

https://bugs.kali.org/view.php?id=5120#c10014

**Comment by ethicalhack3r** [comment]
TLDR; If using Kali do not install wpscan from any other source (such as `gem install wpscan`), use Kali's (`apt-get update` I assume)

**Comment by sumgro** [comment]
Here is what I did to resolve this (based on the suggestion)

Remove wpscan installed manually through gem - gem uninstall wpscan
Install wpscan via Kali repository - apt get wpscan

wpscan now works.

**Comment by sumgro** [comment]
> TLDR; If using Kali do not install wpscan from any other source (such as `gem install wpscan`), use Kali's (`apt-get update` I assume)

True that...

**Comment by Rzqu** [comment]
I also have this issue so im following this thread.

**Comment by Rzqu** [comment]
> Here is what I did to resolve this (based on the suggestion)
> 
> Remove wpscan installed manually through gem - gem uninstall wpscan
> Install wpscan via Kali repository - apt get wpscan
> 
> wpscan now works.

This does not work for me.

**Comment by sumgro** [comment]
@rzqu this seems like a different error... maybe @ethicalhack3r can confirm

**Comment by Rzqu** [comment]
@sumgro After re-installing wpscan through gem i get this

![image](https://user-images.githubusercontent.com/39829605/49085523-1c3c0f80-f24a-11e8-8b2a-5cc4273f9cd5.png)

Installing it through apt-get install does nothing for me, the install goes through but when running wpscan i end up with this 

![image](https://user-images.githubusercontent.com/39829605/49083945-0298c900-f246-11e8-87fe-f76b47e6ec4e.png)

**Comment by sumgro** [comment]
@rzqu could try removing wpscan through both gem and apt, use:
gem uninstall wpscan
apt remove wpscan

Then, install it via apt using: apt install wpscan

**Comment by Rzqu** [comment]
@sumgro Ive tried what you suggested and im now faced with this error

![image](https://user-images.githubusercontent.com/39829605/49091687-472d6000-f258-11e8-9b0e-b530d04064db.png)

**Comment by cohernandez** [comment]
I solved the same problem with this sequence 
1. gem uninstall  wpscan
2. apt remove wpscan
3. apt-get autoremove
4. apt-get autoclean
5. apt install wpscan

**Comment by dhd0004** [comment]
I did this sequence:
gem uninstall wpscan
apt remove wpscan
apt-get autoremove
apt-get autoclean
apt install wpscan


And still having the same error:
/usr/lib/ruby/2.5.0/rubygems/specification.rb:1466:in `block in activate_dependencies': can't satisfy 'activesupport (~> 5.2)', already activated 'activesupport-4.2.10' (Gem::LoadError)

**Comment by Rzqu** [comment]
@dhd0004 have a look at this https://bugs.kali.org/view.php?id=5134

Apparently mine was leaving little gems behind (excuse the pun)

i renamed /var/lib/gems/2.5.0/gems to /var/lib/gems/2.5.0/gems.old and my WPScan started working.

Maybe try that?

**Comment by dhd0004** [comment]
that did not word for me:

> ls -lsrt /var/lib/gems/2.5.0/gems.old/
> total 36
> 4 drwxr-xr-x 4 root root 4096 Nov 15 16:59 concurrent-ruby-1.1.3
> 4 drwxr-xr-x 3 root root 4096 Nov 15 16:59 activesupport-5.1.6
> 4 drwxr-xr-x 3 root root 4096 Nov 15 16:59 opt_parse_validator-0.0.15.2
> 4 drwxr-xr-x 5 root root 4096 Nov 15 16:59 cms_scanner-0.0.39.0
> 4 drwxr-xr-x 3 root root 4096 Nov 15 17:00 activesupport-5.2.1
> 4 drwxr-xr-x 3 root root 4096 Nov 15 17:00 ruby-progressbar-1.10.0
> 4 drwxr-xr-x 9 root root 4096 Nov 15 17:00 yajl-ruby-1.4.1
> 4 drwxr-xr-x 4 root root 4096 Nov 15 17:00 cms_scanner-0.0.41.0
> 4 drwxr-xr-x 4 root root 4096 Nov 15 17:01 cms_scanner-0.0.40.3
> 
Y renamed from this:

> ls -lsrt /var/lib/gems/2.5.0/gems
> ls: cannot access '/var/lib/gems/2.5.0/gems': No such file or directory

Error still persists:

> wpscan
> Traceback (most recent call last):
> 	11: from /usr/local/bin/wpscan:23:in `<main>'
> 	10: from /usr/lib/ruby/2.5.0/rubygems.rb:309:in `activate_bin_path'
> 	 9: from /usr/lib/ruby/2.5.0/rubygems.rb:309:in `synchronize'
> 	 8: from /usr/lib/ruby/2.5.0/rubygems.rb:310:in `block in activate_bin_path'
> 	 7: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in `activate'
> 	 6: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `activate_dependencies'
> 	 5: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `each'
> 	 4: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1472:in `block in activate_dependencies'
> 	 3: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in `activate'
> 	 2: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `activate_dependencies'
> 	 1: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `each'
> /usr/lib/ruby/2.5.0/rubygems/specification.rb:1466:in `block in activate_dependencies': can't satisfy 'activesupport (~> 5.2)', already activated 'activesupport-4.2.10' (Gem::LoadError)
>

**Comment by Rzqu** [comment]
@dhd0004 

Im assuming that youve already done gem uninstall wpscan before renaming that folder?

**Comment by dhd0004** [comment]
yes, i executed these commands:

> gem uninstall wpscan
> sudo apt-get autoremove wpscan
> sudo mv /var/lib/gems/2.5.0/gems /var/lib/gems/2.5.0/gems.old
> sudo apt-get install wpscan
> wpscan
> Traceback (most recent call last):
> 	11: from /usr/local/bin/wpscan:23:in `<main>'
> 	10: from /usr/lib/ruby/2.5.0/rubygems.rb:309:in `activate_bin_path'
> 	 9: from /usr/lib/ruby/2.5.0/rubygems.rb:309:in `synchronize'
> 	 8: from /usr/lib/ruby/2.5.0/rubygems.rb:310:in `block in activate_bin_path'
> 	 7: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in `activate'
> 	 6: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `activate_dependencies'
> 	 5: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `each'
> 	 4: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1472:in `block in activate_dependencies'
> 	 3: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1440:in `activate'
> 	 2: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `activate_dependencies'
> 	 1: from /usr/lib/ruby/2.5.0/rubygems/specification.rb:1458:in `each'
> /usr/lib/ruby/2.5.0/rubygems/specification.rb:1466:in `block in activate_dependencies': can't satisfy 'activesupport (~> 5.2)', already activated 'activesupport-4.2.10' (Gem::LoadError)
>

**Comment by Milimas** [comment]
i typed "gem install wpscan" in terminal and i open wpscan and it works

**Comment by faudbispa** [comment]
I followed this serial, but not working
1.  gem uninstall  wpscan

2.  apt remove wpscan

3.  apt-get autoremove

4.  apt-get autoclean

5.  apt install wpscan

Traceback (most recent call last):
    25: from /usr/bin/wpscan:23:in `<main>'
    24: from /usr/bin/wpscan:23:in `load'
    23: from /usr/share/rubygems-integration/all/gems/wpscan-3.4.4/bin/wpscan:3:in `<top (required)>'
    22: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
    21: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
    20: from /usr/share/rubygems-integration/all/gems/wpscan-3.4.4/lib/wpscan.rb:5:in `<top (required)>'
    19: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
    18: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
    17: from /usr/lib/ruby/vendor_ruby/active_support/all.rb:5:in `<top (required)>'
    16: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
    15: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
    14: from /usr/lib/ruby/vendor_ruby/active_support/core_ext.rb:3:in `<top (required)>'
    13: from /usr/lib/ruby/vendor_ruby/active_support/core_ext.rb:3:in `each'
    12: from /usr/lib/ruby/vendor_ruby/active_support/core_ext.rb:4:in `block in <top (required)>'
    11: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
    10: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
     9: from /usr/lib/ruby/vendor_ruby/active_support/core_ext/big_decimal.rb:3:in `<top (required)>'
     8: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
     7: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
     6: from /usr/lib/ruby/vendor_ruby/active_support/core_ext/big_decimal/conversions.rb:4:in `<top (required)>'
     5: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
     4: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
     3: from /var/lib/gems/2.5.0/gems/bigdecimal-1.4.3/lib/bigdecimal/util.rb:9:in `<top (required)>'
     2: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:39:in `require'
     1: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `rescue in require'
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `require': /var/lib/gems/2.5.0/gems/bigdecimal-1.4.3/lib/bigdecimal/util.so: undefined symbol: rmpd_util_str_to_d - /var/lib/gems/2.5.0/gems/bigdecimal-1.4.3/lib/bigdecimal/util.so (LoadError)

**Comment by afewcc** [comment]
@dhd0004  and @faudbispa 

Have you tried installing these 2 gems

`gem install activesupport`
`gem install public_suffix`

and then running wpscan ?

**Comment by muaz-alhilal** [comment]
same issue with me 
``Traceback (most recent call last):
	28: from /usr/bin/wpscan:23:in `<main>'
	27: from /usr/bin/wpscan:23:in `load'
	26: from /usr/share/rubygems-integration/all/gems/wpscan-3.4.4/bin/wpscan:3:in `<top (required)>'
	25: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	24: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	23: from /usr/share/rubygems-integration/all/gems/wpscan-3.4.4/lib/wpscan.rb:5:in `<top (required)>'
	22: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	21: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	20: from /var/lib/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/all.rb:5:in `<top (required)>'
	19: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	18: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	17: from /var/lib/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext.rb:3:in `<top (required)>'
	16: from /var/lib/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext.rb:3:in `each'
	15: from /var/lib/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext.rb:4:in `block in <top (required)>'
	14: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	13: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	12: from /var/lib/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext/object.rb:13:in `<top (required)>'
	11: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	10: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 9: from /var/lib/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext/object/json.rb:8:in `<top (required)>'
	 8: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 7: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 6: from /var/lib/gems/2.5.0/gems/activesupport-5.2.2/lib/active_support/core_ext/big_decimal/conversions.rb:4:in `<top (required)>'
	 5: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 4: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 3: from /var/lib/gems/2.5.0/gems/bigdecimal-1.4.3/lib/bigdecimal/util.rb:9:in `<top (required)>'
	 2: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:39:in `require'
	 1: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `rescue in require'
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `require': /var/lib/gems/2.5.0/gems/bigdecimal-1.4.3/lib/bigdecimal/util.so: undefined symbol: rmpd_util_str_to_d - /var/lib/gems/2.5.0/gems/bigdecimal-1.4.3/lib/bigdecimal/util.so (LoadError)

**Comment by erwanlr** [comment]
@muaz-alhilal, @faudbispa  see https://github.com/wpscanteam/wpscan/issues/1283

Closing this issue as reports are no longer related to the main post.

Furthermore, before posting a new issue, please check the Readme, do some research either in the issues here or tracker of Kali, and google.

**Comment by faudbispa** [comment]
Sir,
My prob was solved and I informed it by mail. And I gave the solution too.
Thank you

**Comment by soham7** [comment]
wpscan

Traceback (most recent call last):
	25: from /usr/bin/wpscan:23:in `<main>'
	24: from /usr/bin/wpscan:23:in `load'
	23: from /usr/share/rubygems-integration/all/gems/wpscan-3.4.5/bin/wpscan:3:in `<top (required)>'
	22: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	21: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	20: from /usr/share/rubygems-integration/all/gems/wpscan-3.4.5/lib/wpscan.rb:5:in `<top (required)>'
	19: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	18: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	17: from /var/lib/gems/2.5.0/gems/activesupport-5.2.3/lib/active_support/all.rb:5:in `<top (required)>'
	16: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	15: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	14: from /var/lib/gems/2.5.0/gems/activesupport-5.2.3/lib/active_support/core_ext.rb:3:in `<top (required)>'
	13: from /var/lib/gems/2.5.0/gems/activesupport-5.2.3/lib/active_support/core_ext.rb:3:in `each'
	12: from /var/lib/gems/2.5.0/gems/activesupport-5.2.3/lib/active_support/core_ext.rb:4:in `block in <top (required)>'
	11: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	10: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 9: from /var/lib/gems/2.5.0/gems/activesupport-5.2.3/lib/active_support/core_ext/big_decimal.rb:3:in `<top (required)>'
	 8: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 7: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 6: from /var/lib/gems/2.5.0/gems/activesupport-5.2.3/lib/active_support/core_ext/big_decimal/conversions.rb:4:in `<top (required)>'
	 5: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 4: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:59:in `require'
	 3: from /var/lib/gems/2.5.0/gems/bigdecimal-1.4.3/lib/bigdecimal/util.rb:9:in `<top (required)>'
	 2: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:39:in `require'
	 1: from /usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `rescue in require'
/usr/lib/ruby/2.5.0/rubygems/core_ext/kernel_require.rb:135:in `require': /var/lib/gems/2.5.0/gems/bigdecimal-1.4.3/lib/bigdecimal/util.so: undefined symbol: rmpd_util_str_to_d - /var/lib/gems/2.5.0/gems/bigdecimal-1.4.3/lib/bigdecimal/util.so (LoadError)


Tried every cure but this seems to be the last scene. Can anybody please help?

**Comment by xffxe4** [comment]
Resolved the same issue. In my fix, I did NOT uninstall anything. 

apt-get install ruby-dev
gem install ffi --platform=ruby
gem install yajl-ruby
apt-get install libxslt-dev libxml2-dev
gem install nokogiri -- --use-system-libraries

Then wpscan worked.

**Comment by sergioab7** [comment]
@xffxe4 you are my hero, thanks.

**Comment by EmperorSrikanth** [comment]
Reall

> Resolved the same issue. In my fix, I did NOT uninstall anything.
> 
> apt-get install ruby-dev
> gem install ffi --platform=ruby
> gem install yajl-ruby
> apt-get install libxslt-dev libxml2-dev
> gem install nokogiri -- --use-system-libraries
> 
> Then wpscan worked.

Yes it worked for me too!

**Comment by PanDa1G1** [comment]
> gem install nokogiri -- --use-system-libraries

it's useful,thank you!

**Comment by gsandovalbonilla** [comment]
Hi,
I solved the issue with this comand agpt-get upgrade.
I did two time and it works

**Comment by xrobb30** [comment]
> I solved the same problem with this sequence
> 
>     1. gem uninstall  wpscan
> 
>     2. apt remove wpscan
> 
>     3. apt-get autoremove
> 
>     4. apt-get autoclean
> 
>     5. apt install wpscan

i did this step, and its workedd thank you

**Comment by Ehsan-Nezami** [comment]
use the following command to fix that:

**Comment by HarshitJosh1** [comment]
Was having same issue 

Try gem install wpscan

worked on my side

**Comment by BalbeerJaat** [comment]
gem update
gem install wpscan
wpscan --update

Solved it for me!

**Comment by 0xBones** [comment]
sudo gem install wpscan 

worked for me.

Kali 2020.1a

**Comment by spider8892** [comment]
> Was having same issue
> 
> Try - gem install wpscan
> 
> worked on my side

Thanks, problem solved

**Comment by nasrallah-hacker** [comment]
> first, uninstall all Wpscan with the following command:
> 
> `gem uninstall wpscan`
> `apt remove wpscan`
> 
> then, use the following command to fix it:
> 
> `sudo apt-get update`
> `gpg --keyserver hkp://keys.gnupg.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB`
> `\curl -sSL https://get.rvm.io | bash -s stable --ruby --ignore-dotfiles`
> `\curl -sSL https://get.rvm.io | bash -s stable --rails`
> `gem install rails 5`
> `source /usr/local/rvm/scripts/rvm`
> `rvm reinstall ruby 2.6.0`
> `rvm install "ruby-2.5.3"`
> `gem install wpscan`
> 
> Make sure it is installed correctly with this command:
> 
> `which ruby`
> `ruby --version`
> `rails -v`
> 
> It works now.

it work for me 
thank you

**Comment by zick43** [comment]
I only use ¨gem install wpscan¨ and work for me

**Comment by RoyeeW** [comment]
> 
apt-get remove --purge wpscan
apt autoremove 
apt-get update
apt install wpscan
--------
work for me

**Comment by AbdullahAlSolaiman** [comment]
> apt-get remove --purge wpscan

Worked for me as well

**Comment by Zuprakk** [comment]
this is how i solved the problem
apt-get install wpscan
